pkgs/applications/virtualization/containerd/1.4.nix

```
@@ -0,0 +1,56 @@
+{ lib
+, fetchFromGitHub
+, buildGoPackage
+, btrfs-progs
+, go-md2man
+, installShellFiles
+, util-linux
+, nixosTests
+}:
+
+buildGoPackage rec {
+  pname = "containerd";
+  version = "1.4.9";
+
+  src = fetchFromGitHub {
+    owner = "containerd";
+    repo = "containerd";
+    rev = "v${version}";
+    sha256 = "1ykikks6ihgg899ibk9m9m0hqrbss0cx7l7z4yjb873b10bacj52";
+  };
+
+  goPackagePath = "github.com/containerd/containerd";
+  outputs = [ "out" "man" ];
+
+  nativeBuildInputs = [ go-md2man installShellFiles util-linux ];
+
+  buildInputs = [ btrfs-progs ];
+
+  buildFlags = [ "VERSION=v${version}" "REVISION=${src.rev}" ];
+
+  BUILDTAGS = [ ]
+    ++ lib.optional (btrfs-progs == null) "no_btrfs";
```

```suggestion
  BUILDTAGS = lib.optional (btrfs-progs == null) "no_btrfs";
```
Also overwriting to null is not really supported.

---

I restored this from the pre 1.5.1 derivation of containerd. Since the 1.4.x derivation is meant only for docker I can just drop this entirely, I think?

---

I am not sure but if docker uses no btrfs at all.

---

I think you can freely drop this, because `btrfs-tools` being missing from NixOS installations being able to run `moby` / dockerd don't really seem to exist.

---

Dropped the conditional.